### PR TITLE
Per 9572 use dialog cdk

### PR DESCRIPTION
**Pull request by crisnicandrei**
This pull request replaces the previous dialog module from the web app with angular's cdk module.

### CODE REVIEW

**Comment by crisnicandrei** [comment]
@meisekimiu i think i have fixed the routed wrapper component, but I would still like your input on the code before I can mark this as ready to review, because I still need to change some designs and add some tests. Thanks!

**Comment by crisnicandrei** [comment]
@meisekimiu my main issues was: `    this.dialogRef = this.dialog.createDialog(
      this.dialogToken,
      dialogData,
      this.dialogOptions,
      this.outletTemplate,
      this.route,
    );

    this.dialogRef.dialogComponent.show();

    this.dialogRef.closePromise.finally(() => {
      this.dialogRef = null;
      if (!this.closedByNavigate) {
        const targetRoute = this.routeHistory.previousRoute || '/app/';
        this.router.navigateByUrl(targetRoute);
      }
    });` because i was trying to initialize the ref using the old method whilst calling dialog.open from the new method, but then I switched it all to use the cdk's methods and it worked

File: src/app/app.routes.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export interface RouteData {
   isPublic?: boolean;
 
   dialogToken?: DialogComponentToken;
+  component?: any;
```
Comment (meisekimiu) [inline src/app/app.routes.ts:29]: Does this need to be `any` type? I feel like this should match the type that is allowed by the CDK Dialog `open` method. Also the `Component` type included at the top is currently unused in this file.

File: src/app/app.routes.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export interface RouteData {
   isPublic?: boolean;
 
   dialogToken?: DialogComponentToken;
```
Comment (meisekimiu) [inline src/app/app.routes.ts:28]: I believe if we fully switch over to the CDK Dialog system, this property should be unused. We should aim to get this property in a removable (and then removed) state.

File: src/app/core/components/public-settings/public-settings.component.spec.ts [hunk 3]
```diff
@@ -32,14 +32,15 @@ const mockApiService = {
 
 describe('PublicSettingsComponent', () => {
   let shallow: Shallow<PublicSettingsComponent>;
+  let messageShown = false;
```
Comment (meisekimiu) [inline src/app/core/components/public-settings/public-settings.component.spec.ts:35]: Is this value used in tests at all? Its value doesn't seem to be reset in between test runs if it is used.

File: src/app/file-browser/components/edit-tags/edit-tags.component.spec.ts [hunk 4]
```diff
@@ -198,21 +198,6 @@ describe('EditTagsComponent', () => {
...
-    fixture.detectChanges();
-
-    find('.manage-tags-message .manage-tags-link').triggerEventHandler(
-      'click',
-      {},
-    );
-
-    expect(dialogOpenSpy.open).toHaveBeenCalled();
-  });
-
```
Comment (meisekimiu) [inline src/app/file-browser/components/edit-tags/edit-tags.component.spec.ts:215]: I think this is an example of a deleted test that didn't need to be fully deleted. If I comment out or remove the dialog call completely, tests will still pass. Do we really want this behavior to not be covered by tests?

File: src/app/shared/components/account-dropdown/account-dropdown.component.ts [hunk 5]
```diff
@@ -24,11 +24,19 @@ import {
...
   animateChild,
 } from '@angular/animations';
-import { Dialog } from '@root/app/dialog/dialog.module';
-import { SettingsTab } from '@core/components/account-settings-dialog/account-settings-dialog.component';
 import { GuidedTourService } from '@shared/services/guided-tour/guided-tour.service';
 import { GuidedTourEvent } from '@shared/services/guided-tour/events';
 import { EventService } from '@shared/services/event/event.service';
+import { ApiService } from '@shared/services/api/api.service';
+import { DeviceService } from '@shared/services/device/device.service';
+import { AnalyticsService } from '@shared/services/analytics/analytics.service';
```
Comment (meisekimiu) [inline src/app/shared/components/account-dropdown/account-dropdown.component.ts:32]: Are these new imports used?

File: src/app/shared/components/routed-dialog-wrapper/routed-dialog-wrapper.component.ts [hunk 6]
```diff
@@ -1,96 +1,74 @@
...
+    const dialogToken = (this.route.snapshot.data as RouteData).dialogToken;
+    const component = (this.route.snapshot.data as RouteData).component;
 
-    if (!this.dialogToken) {
+    if (!dialogToken) {
       throw new Error(
-        "RoutedDialogWrapperComponent - missing dialog token on route data, can't open dialog",
+        "RoutedDialogWrapperComponent - missing dialog token on route data, can't open dialog"
       );
     }
```
Comment (meisekimiu) [inline src/app/shared/components/routed-dialog-wrapper/routed-dialog-wrapper.component.ts:61]: As far as I understand it, the DialogToken type should be completely unused at this point. This means we should be able to remove these lines checking for DialogToken completely, including this throw statement.

**Review by meisekimiu — CHANGES_REQUESTED** [review]
I think this still needs a bit of work. I've found some components aren't displaying properly such as the Welcome dialog that pops up when a user finishes onboarding, and the Archive Profile screen as well.

The old Dialog code is also still intact. While it isn't being used anymore, it is still imported in a few places. It should be completely removed at this point, since the whole point of this work is to replace it entirely. Once we've completely replaced an old piece of code there is no point in keeping it around.

**Comment by meisekimiu** [comment]
Can you try rebasing this branch onto the current version of main?

**Comment by crisnicandrei** [comment]
@meisekimiu thank you! It seems codecov/patch is sending some warning to some lines that were here before

**Review by meisekimiu — COMMENTED** [review]
Can you improve the test coverage a bit? The patch coverage change is down about a half of a percent and that is going to block merging this. I pointed out an example of a test that didn't have to be fully deleted in my previous review comments.

**Comment by crisnicandrei** [comment]
@meisekimiu thanks! I will!

File: src/app/views/components/timeline-view/timeline-view.component.ts [hunk 7]
```diff
@@ -152,23 +153,21 @@ export class TimelineViewComponent implements OnInit, AfterViewInit, OnDestroy {
   };
   constructor(
     @Inject(DIALOG_DATA) public dialogData: any,
-    @Inject(OUTLET_TEMPLATE) public outletTemplate: any,
+    // @Inject(OUTLET_TEMPLATE) public outletTemplate: any,
```
Comment (meisekimiu) [inline src/app/views/components/timeline-view/timeline-view.component.ts:156]:
```suggestion
```
Let's not commit commented out code.

File: src/app/shared/components/routed-dialog-wrapper/routed-dialog-wrapper.component.ts [hunk 8]
```diff
@@ -97,4 +66,24 @@ export class RoutedDialogWrapperComponent
...
+    };
+
+    this.dialogRef = this.dialogService.open(component, config);
+
+    this.dialogRef.closed.subscribe(() => {
+      this.dialogRef = null;
+      if (!this.closedByNavigate) {
+        const targetRoute = this.routeHistory.previousRoute || '/app/';
+        this.router.navigateByUrl(targetRoute);
+      }
```
Comment (meisekimiu) [inline src/app/shared/components/routed-dialog-wrapper/routed-dialog-wrapper.component.ts:86]: While sometimes these codecov warnings can be ignored, this is a good example of a place where test coverage would have been useful as part of the refactoring process. If I were restructuring this specific bit of code, I would have written tests to cover the original routed dialog wrapper *before* doing any CDK Dialog conversions. Once these were written and passed, *then* I would begin refactoring. In an ideal world, once we restructure the RoutedDialogWrapper the tests would still pass proving that our refactoring didn't break anything. In practice, we'd probably have to edit the tests to match the new API of the CDK Dialog. But I do recommend writing some tests up front before doing actual restructuring, so that they [characterize  the preexisting behavior](https://en.wikipedia.org/wiki/Characterization_test) and we can verify that behavior hasn't changed.

**Review by meisekimiu — APPROVED** [review]
This looks mostly good, though I do have some advice for future refactoring PRs. Ideally a refactoring should have no effect on functionality (and no negative effect on test coverage). One of the main benefits of unit tests is that they enable [refactoring](https://refactoring.com/)  by allowing us to make modifications to the code and verify that we have not altered any behavior or introduced any bugs in the process.

I highly recommend for future refactoring work to work in small steps (converting over a component at a time, for example) and running tests after each of these small adjustments so you can confirm nothing is broken. To me it seems like all component conversions were done first and then tests were adjusted to pass again after the fact. The advantage to running tests frequently after each change is that you can quickly identify *what* broke tests. It's a lot harder when you have many failing tests.

I will approve this, though I recommend having a manual QA pass at this as well.

**Comment by meisekimiu** [comment]
Adding @k8lyn6 onto this. The PR can be manually tested by verifying that none of the modals in the app are broken.

**Comment by k8lyn6** [comment]
@emlesienk to QA this, go through and open all the modals across the app (the new windows that open on top of workspaces). These pop open after clicking most buttons and menu items. Let us know if there are any that appear broken.

**Review by emlesienk — COMMENTED** [review]
TLDR; Notifications and "add connection" aren't opening properly. The rest of the issues seem to be formatting issues that cause the designs to look too tight (see screenshots).
- The notifications module isn't opening properly
- Items in the account menu show a box around the x in the corner 
- Won't open the "add connection" in the Connections module
- The "pending archive" box is off, design is cut off / not enough space. Same deal with the "share" function.
- Publish option screen looks different than what's on prod


menu_screenshot
publish_screenshot
share_screenshot
add_connection_screenshot
pending_connections_screenshot
storage_screenshot
notifications_screenshot

**Comment by k8lyn6** [comment]
@emlesienk can you re-review?

**Review by emlesienk — APPROVED** [review]
Looks good, thanks Andrei!
